File: src/console/command/parameter/def/TemplateParameter.php
<?php

namespace pocketcloud\cloud\console\command\parameter\def;

use pocketcloud\cloud\console\command\parameter\CommandParameter;
use pocketcloud\cloud\console\command\parameter\exception\ArgumentParseException;
use pocketcloud\cloud\template\Template;
use pocketcloud\cloud\template\TemplateManager;

readonly class TemplateParameter extends CommandParameter {

    public function parseValue(string $input): Template {
        if (($template = TemplateManager::getInstance()->get($input)) !== null) return $template;
        throw new ArgumentParseException();
    }

    public function onTabCompleteMatch(string $currentArg): array {
        return array_filter(array_keys(TemplateManager::getInstance()->getAll()), fn(string $template) => str_contains(strtolower($template), $currentArg));
    }

    public function getType(): string {
        return "template";
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/console/command/parameter/def/TemplateParameter.java
package de.pocketcloud.cloud.console.command.parameter.def;

import de.pocketcloud.api.component.template.ITemplate;
import de.pocketcloud.cloud.PocketCloud;
import de.pocketcloud.cloud.console.command.exception.ArgumentParseException;
import de.pocketcloud.cloud.console.command.parameter.BaseCommandParameter;

import java.util.List;

public final class TemplateParameter extends BaseCommandParameter {

    public TemplateParameter(String name, boolean optional) {
        super(name, optional);
    }

    public TemplateParameter(String name, boolean optional, String customErrorMessage) {
        super(name, optional, customErrorMessage);
    }

    @Override
    public Object parseValue(String input) throws ArgumentParseException {
        return PocketCloud.instance().templates().get(input).orElseThrow(() -> new ArgumentParseException("Template not found"));
    }

    @Override
    public List<String> onTabCompleteMatch(String currentArg) {
        return PocketCloud.instance().templates().getAll().stream().map(ITemplate::name).toList();
    }

    @Override
    public String getType() {
        return "template";
    }
}